src/Form/DonationRequestType.php

```diff
@@ -68,6 +87,19 @@ public function buildView(FormView $view, FormInterface $form, array $options)
         }
     }
 
+    public function preSetData(FormEvent $formEvent): void
+    {
+        if (!$formEvent->getData()) {
+            $user = $this->tokenStorage->getToken()->getUser();
+
+            if (!$user instanceof Adherent) {
+                $user = null;
+            }
+
+            $formEvent->setData($this->donationRequestUtils->createFromRequest($this->requestStack->getCurrentRequest(), $user));
```

Comment:
why do you this?

Reply:
Because it's in preset data we have to init default value if we need to use a factory. If not the form type will try to instantiate 'data_class' and will throw an exception because of missing parameters.